fixit/rules/use_assert_is_not_none.py

```diff
@@ -76,173 +86,86 @@ class UseAssertIsNotNoneRule(CstLintRule):
-                            comparisons=[
-                                m.ComparisonTarget(m.IsNot(), comparator=m.Name("None"))
-                            ]
-                        )
-                    )
-                ],
-            ),
-        ):
-            new_call = node.with_changes(
-                func=cst.Attribute(
-                    value=cst.Name("self"), attr=cst.Name("assertIsNotNone")
+                func=m.Attribute(
+                    value=m.Name("self"),
+                    attr=m.SaveMatchedNode(
+                        m.OneOf(m.Name("assertTrue"), m.Name("assertFalse")),
+                        "assertion_name",
+                    ),
                 ),
-                args=[cst.Arg(ensure_type(node.args[0].value, cst.Comparison).left)],
-            )
-            self.report(node, replacement=new_call)
-
-        # `self.assertTrue(not x is None)` -> `self.assertIsNotNone(x)`
-        elif m.matches(
-            node,
-            m.Call(
-                func=m.Attribute(value=m.Name("self"), attr=m.Name("assertTrue")),
                 args=[
                     m.Arg(
-                        value=m.UnaryOperation(
-                            operator=m.Not(),
-                            expression=m.Comparison(
-                                comparisons=[
-                                    m.ComparisonTarget(
-                                        m.Is(), comparator=m.Name("None")
-                                    )
-                                ]
+                        m.SaveMatchedNode(
+                            m.OneOf(
+                                m.Comparison(
+                                    comparisons=[
+                                        m.ComparisonTarget(
+                                            m.SaveMatchedNode(
+                                                m.OneOf(m.IsNot(), m.Is()),
+                                                "comparison_type",
+                                            ),
+                                            comparator=m.Name("None"),
+                                        )
+                                    ]
+                                ),
+                                m.UnaryOperation(
+                                    operator=m.Not(),
+                                    expression=m.Comparison(
+                                        comparisons=[
+                                            m.ComparisonTarget(
+                                                m.SaveMatchedNode(
+                                                    m.OneOf(m.Is(), m.IsNot()),
+                                                    "comparison_type",
+                                                ),
+                                                comparator=m.Name("None"),
+                                            )
+                                        ]
+                                    ),
```

Comment:
The entire `m.Comparison(...)` is the same as the one used above.
We can move it as a variable (e.g. `match_compare_is_none = m.Comparison(...)`) and use the variable in both places  to remove the duplication.